relay/sources/rudder-package/src/versions.rs

```diff
@@ -0,0 +1,231 @@
+                }
+            }
+        };
+        // Test if beta
+        let _ = match beta_regex.captures(s) {
+            None => (),
+            Some(c) => {
+                let version = c["version"].to_string().parse().unwrap();
+                if is_nightly {
+                    return Ok(RudderVersionMode::Beta { version })
+                } else {
+                    return Ok(RudderVersionMode::BetaNightly { version })
+                }
+            }
+        };
+        // Test if rc
+        let _ = match rc_regex.captures(s) {
+            None => (),
+            Some(c) => {
+                let version = c["version"].to_string().parse().unwrap();
+                if is_nightly {
+                    return Ok(RudderVersionMode::Rc { version })
+                } else {
+                    return Ok(RudderVersionMode::RcNightly { version })
+                }
+            }
+        };
+        // Test if git
+        let _ = match git_regex.captures(s) {
+            None => (),
+            Some(c) => {
+                let version = c["version"].to_string().parse().unwrap();
+                return Ok(RudderVersionMode::Git { version })
+            }
+        };
+        bail!("Unparsable Rudder version mode '{}'", s)
+    }
+}
+
+ #[derive(PartialEq, Debug)]
+struct RudderVersion {
+    pub major: u16,
+    pub minor: u16,
+    pub patch: u16,
+    pub mode: RudderVersionMode,
+    pub raw_version: String,
+}
+
+
+impl FromStr for RudderVersion {
+    type Err =  Error;
+
+    fn from_str(raw: &str) -> Result<Self, Self::Err> {
+        let re = Regex::new(r"^(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(?<mode>.*)$")?;
+        let caps = match re.captures(raw) {
+            None => bail!("Unparsable rudder version '{}'", raw),
+            Some(c) => c
+        };
+        let major: u16 = u16::from_str(&caps["major"].to_string())?;
+        let minor: u16 = u16::from_str(&caps["minor"].to_string())?;
+        let patch: u16 = u16::from_str(&caps["patch"].to_string())?;
+        let mode: RudderVersionMode = RudderVersionMode::from_str(&caps["mode"])?;
+        let raw_version = String::from(raw);
+
+        Ok(RudderVersion {
+            major,
+            minor,
+            patch,
+            mode,
+            raw_version
+        })
+    }
+}
+
+ #[derive(PartialEq, Debug)]
+struct PluginVersion {
+    pub major: u32,
+    pub minor: u32,
+    pub nightly: bool
+}
+impl FromStr for PluginVersion {
+    type Err =  Error;
+
+    fn from_str(raw: &str) -> Result<Self, Self::Err> {
+        let is_nightly = Regex::new(r".*-nightly$")?.is_match(raw);
+        let re = Regex::new(r"^(?<major>\d+)\.(?<minor>\d+).*$")?;
+        let caps = match re.captures(raw) {
+            None => bail!("Unparsable plugin version '{}'", raw),
+            Some(c) => c
+        };
+        let major: u32 = u32::from_str(&caps["major"].to_string())?;
+        let minor: u32 = u32::from_str(&caps["minor"].to_string())?;
```

Comment:
This is fine but I think it could be simpler. You probably don't need the conversion to an owned string in the middle, not to specify the type twice, and might be able to do something like:

```rust
let minor: u32 = caps["major"].parse()?;
let minor = caps["major"].parse::<u32>()?;
```